sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -1022,6 +1022,30 @@ public static GpuColumnVector[] extractColumns(Table table, DataType[] colType)
     }
   }
 
+  /**
+   * Convert the table into host columns and return them, outside of a ColumnarBatch.
+   * @param colType the types of the columns.
+   */
+  public static RapidsHostColumnVector[] extractHostColumns(Table table, DataType[] colType) {
+    try (ColumnarBatch batch = from(table, colType)) {
+      GpuColumnVector[] gpuCols = extractColumns(batch);
+      RapidsHostColumnVector[] hostCols = new RapidsHostColumnVector[gpuCols.length];
+      try {
+        for (int i = 0; i < gpuCols.length; i++) {
+          hostCols[i] = gpuCols[i].copyToHost();
+        }
+      } catch (Exception e) {
+        for (RapidsHostColumnVector hostCol : hostCols) {
+          if (hostCol != null) {
+            hostCol.close();
```

Comment:
Each `close` call should be catched.
```suggestion
            try {
              hostCol.close();
            } catch (Exception e) { /* ignore */ }
```